roles/kubevirt/tasks/provision.yml

```diff
@@ -8,126 +8,52 @@
-      when: byo_demo_content.stat.exists == False and offline_demo_content.stat.exists == False
-
-    - name: Copy Offline Demo Content to /tmp
-      copy:
-        src: "{{ offline_template_dir }}/v{{ version }}/demo-content.yaml"
-        dest: "/tmp/demo-content.yaml"
-      when: offline_demo_content is not skipped and offline_demo_content.stat.exists == True
-
-    - name: Copy BYO Demo Content to /tmp
-      copy:
-        src: "{{ kubevirt_template_dir }}/demo-content.yaml"
-        dest: "/tmp/demo-content.yaml"
-      when: byo_demo_content.stat.exists == False and (offline_demo_content is skipped or offline_demo_content.stat.exists == False)
-
-    - name: Create Demo Content
-      command: "{{ cluster_command }} apply -f /tmp/demo-content.yaml"
-      when: deploy_demo == true
-
-# VM Templates manifests
-- name: install vm templates manifests
-  block:
-  - name: Check for vm templates in {{ kubevirt_template_dir }}
-    stat:
-      path: "{{ kubevirt_template_dir }}/common-templates.yaml"
-    register: byo_vm_templates
-
-  - name: Apply vm templates from {{ kubevirt_template_dir }}/common-templates.yaml
-    command: "{{ cluster_command }} apply -f {{ kubevirt_template_dir }}/common-templates.yaml"
-    when: byo_vm_templates.stat.exists == true
-
-  - name: Download KubeVirt VM templates
-    block:
-      - name: Download KubeVirt default VM templates for version < 0.6.0
-        get_url:
-          url: "https://raw.githubusercontent.com/kubevirt/kubevirt/v{{ version }}/cluster/examples/{{ item }}.yaml"
-          dest: "/tmp/{{ item }}.yaml"
-        with_items:
-          - "{{ default_vm_templates }}"
-        when: "version is version('0.6.0', '<')"
-
-      - name: Create default VM templates in OpenShift Namespace for version < 0.6.0
-        shell: "oc apply -f /tmp/{{ item }}.yaml -n openshift"
-        with_items:
-          - "{{ default_vm_templates }}"
-        when: "version is version('0.6.0', '<')"
-
-      - name: Download KubeVirt default VM templates for version >= 0.6.0
-        get_url:
-          url: "https://github.com/kubevirt/common-templates/releases/download/v{{ common_templates_version }}/common-templates-v{{ common_templates_version }}.yaml"
-          dest: "/tmp/common-templates-v{{ common_templates_version }}.yaml"
-        when: "version is version('0.6.0', '>=')"
-
-      - name: Create default VM templates in OpenShift Namespace for version >= 0.6.0
-        shell: "oc apply -f /tmp/common-templates-v{{ common_templates_version }}.yaml -n openshift"
-        when: "version is version('0.6.0', '>=')"
-    when: byo_vm_templates.stat.exists == false
-  when: platform == "openshift"
+  command: "{{ cluster_command }} apply -f /tmp/kubevirt-cr.yaml --validate=false"
+
+- name: Wait until virt-operator is running
+  shell: "{{ cluster_command }} -n {{ namespace }} get deployment | grep virt-operator | awk \'{ if ($3 == $4) print \"0\"; else print \"1\"}\'"
```

Comment:
1. Can't we use `oc get -o jsonpath=...` instead of ` awk?
This playbook assumes that awk is installed, which is not true for all systems.

2. Can we use a loop instead of duplicating the code that verifies that the resources are running?

Reply:
Do you know how to return multiple values from different fields using jsonpath (which is essentially jq)?  I think this check would have to be two lines.

I don't you can use a loop because that does not execute until a condition is satisfied.  Do you have a example of how this would work with a loop?

Reply:
Not sure about returning multiple lines. I think it's better to have 2 lines than to rely on an external tool.

about the loop, I'll try to provide an example, If it didn't work for you, we don't need to block on it.